src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
Instead of `forEach`, use the most suitable array method - `reduce`.
Don’t shorten variables or function names. Check this [guide](https://mate-academy.github.io/style-guides/javascript.html#2311-dont-shorten-variables-or-functions-names).
```suggestion
  const sum = women.reduce((accumulator, woman) => accumulator + woman.died - woman.born, 0);
```